packages/backend/src/graphql/resolvers/wallet_address.ts

```diff
@@ -69,18 +69,20 @@ export const getWalletAddress: QueryResolvers<ApolloContext>['walletAddress'] =
     return walletAddressToGraphql(walletAddress)
   }
 
-export const getWalletAddressByUrl: QueryResolvers<ApolloContext>['walletAddressByUrl'] =
+export const getWalletAddressByUrl: QueryResolvers<TenantedApolloContext>['walletAddressByUrl'] =
   async (
     parent,
     args,
     ctx
   ): Promise<ResolversTypes['WalletAddress'] | null> => {
     const walletAddressService = await ctx.container.use('walletAddressService')
     const walletAddress = await walletAddressService.getByUrl(args.url)
-    return walletAddress ? walletAddressToGraphql(walletAddress) : null
+    return walletAddress && ctx.tenant.id === walletAddress.tenantId
```

Comment:
We want to ensure we are not being 'hacked', the tenant can only access their own wallet addresses.